File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.js
function solution(triangle) {
    for (let d = 2; d<= triangle.length; d++){
        for (var i = 0; i<d; i++){
            if (i == 0) triangle[d-1][i] += triangle[d-2][0];
            else if (i == d-1) triangle[d-1][i] += triangle[d-2][d-2];
            else triangle[d-1][i] += Math.max(triangle[d-2][i],triangle[d-2][i-1]);
        }
    }
    
    return Math.max(...triangle[triangle.length-1]);
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    answer = 0
    l = len(triangle)
    if l == 1: return triangle[0][0]
    
    for h in range(1,l): # 1 2 3 4
        triangle[h][0] += triangle[h-1][0]
        triangle[h][h] += triangle[h-1][h-1]
        for i in range(1,h): #() (1) (1,2) (1,2,3) 
            triangle[h][i] += (max(triangle[h-1][i],triangle[h-1][i-1]))
    return max(triangle[-1])
    